### PR TITLE
Modularize cleanepi

### PR DESCRIPTION
**Pull request by Karim-Mane**
This PR is aimed for improving the implementation of the modules in {cleanepi} (see the package design vignette in the `vignette/` folder).

### CODE REVIEW

File: R/utils.R
Comment (bahadzie) [inline R/utils.R:263]: Suggestion.Since this function is essentially populating a key/value store consider renaming your parameters to reflect it. e.g. `function(x, key value = NULL)`

**Comment by Karim-Mane** [comment]
@bahadzie kindly review the last commit (`5600f0c9`)

File: R/clean_data.R
Comment (bahadzie) [inline R/clean_data.R:69]: `id_col_name = "study_id"`
I think that's what was missing

Comment (Karim-Mane) [reply R/clean_data.R]: Thanks - it works perfectly after correcting this

**Review by bahadzie — APPROVED** [review]
Approving this for now in preparation of `dev day` although I'll be submitting pull requests to this branch as I continue reviewing the individual files and functions until you merge.

**Comment by Karim-Mane** [comment]
@bahadzie - let's merge the `calculate_age` PR into this branch. We will then merge this into main just after. This is because we need the main branch to be up, running with the most up to date version of the package before the development day.

In the mean time, if you have identified others issues, kindly raise a GitHub issue and branch off main after the merge has happened.

**Pull request merged**
